app/Http/Controllers/Manager/WalletController.php

```diff
@@ -328,6 +329,114 @@ public function withdraw(Request $request, AdsRpcClient $rpcClient): JsonRespons
         return $this->withdrawAds($request, $rpcClient);
     }
 
+    private function getSwashVaultBSCAddress(): WalletAddress
+    {
+        try {
+            return new WalletAddress('BSC', config('app.svault_bsc_address'));
+        } catch (InvalidArgumentException $e) {
+            Log::error(sprintf('Invalid BSC address is set: %s', $e->getMessage()));
+            throw new HttpException(Response::HTTP_INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    public function withdrawBatchSwash(Request $request, AdsRpcClient $rpcClient): JsonResponse
+    {
+        /** @var User $user */
+        $user = Auth::user();
+
+        if(!$user->isModerator()){
+            return self::json(['Auth' => 'Only Moderator/Admin can run this API'], JsonResponse::HTTP_FORBIDDEN);
+        }
+
+        $addressFrom = $this->getAdServerAdsAddress();
+        $address = $this->getSwashVaultBSCAddress();
+        $addressTo = $this->getWalletAdsAddress($rpcClient, $address);
+        $message = $this->getWalletAdsMessage($rpcClient, $address);
+        $batchId = hash('sha256', strval(microtime(true)) );
```

Comment:
duplication possibility. If this line is executed twice in the same microsecond, the batch id will be equal!